compiler/runtime/OMRRelocationTarget.hpp

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_RELOCATION_TARGET_INCL
+#define OMR_RELOCATION_TARGET_INCL
+
+#ifndef OMR_RELOCATION_TARGET_CONNECTOR
+#define OMR_RELOCATION_TARGET_CONNECTOR
+namespace OMR { class RelocationTarget; }
+namespace OMR { typedef OMR::RelocationTarget RelocationTargetConnector; }
+#endif
+
+
+#include <stddef.h>
+#include <stdint.h>
+#include "env/TRMemory.hpp"
+#include "env/jittypes.h"
+#include "env/JitConfig.hpp"
+
+namespace TR 
+{
+class RelocationRecord;
+class RelocationRuntime;
+class RelocationTarget;
+}
+
+namespace OMR
+{
+
+class  OMR_EXTENSIBLE RelocationTarget
+   {
+public:
+   TR_ALLOC(TR_Memory::Relocation)
+   void * operator new(size_t, TR::JitConfig *);
+   RelocationTarget(TR::RelocationRuntime *reloRuntime)
+      {
+      _reloRuntime = reloRuntime;
+      }
+   TR::RelocationTarget* self();
```

Comment:
The `self()` function should be `protected`, not `public`.

Reply:
Thank you for the comment! Will address all of them!